torch/csrc/utils/variadic.h

```diff
@@ -84,4 +89,23 @@ inline size_t count_variables(Args&&... args) {
   return CountVariables().apply(std::forward<Args>(args)...).out;
 }
 
+//===----------------------------------------------------------------------===//
+//                std::index_sequence shim for C++11
+//===----------------------------------------------------------------------===//
+
+// A container of type-template parameter indices.
```

Comment:
@ezyang I remember you mentioned in one of your posts in our C++ groups that you were missing index_sequence. Idk if this helps you?

Reply:
Yes, I ended up implementing tuple recursion manually for `TuplePacker`.

Curious about a few things: (1) did this code come from somewhere or was developed from first principles; (2) how does it compare to the C++14 integer sequence, is it exactly compatible but under a different name, or is there porting involved?

Reply:
I came up with this myself (I've done a fair amount of template metaprogramming in the past). `std::index_sequence` will amount to the same thing, but it's implemented far more trivially (you can [see it here](https://github.com/llvm-mirror/libcxx/blob/5b12e3bdb43ac545206501fae955b37cef0c7475/include/utility#L839)) because in C++14 you have templatized typedefs. The core idea is to get a type with a variadic list of integers so that you can grab them with `size_t... Is` in a template parameter list and then use them in a pack expansion